### PR TITLE
Data resource adapter

### PR DESCRIPTION
**Pull request by Mitch391**

### CODE REVIEW

File: tests/unit/ObjectBridgeService/DataResourceAdapterTest.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        $this->objectBridge = $this->createMock(ObjectBridge::class);
+        $this->setUpConvertObjectBridgeConcretesToArrayConcreteInConvertObjectBridgeConcretesToArray();
+    }
+
+    public function testThatEmptyArrayIsReturnedIfNoConcretesAreGivenInConvertObjectBridgeConcretesToArray(): void
+    {
+        $array = $this->adapter->convertObjectBridgeConcretesToArray($this->objectBridge);
+        $this->assertEmpty($array);
+    }
+
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/DataResourceAdapterTest.php]: Preferably, fixture arrangement functionality should be grouped at the top of the test class.

File: tests/unit/ObjectBridgeService/DataResourceAdapterTest.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    /**
+     * @var DataResourceAdapter
+     */
+    private $adapter;
+
+    /**
+     * @var MockObject
+     */
+    private $objectBridge;
+
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/DataResourceAdapterTest.php]: Missing phpdoc

File: tests/unit/ObjectBridgeService/DataResourceAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        $this->objectBridge->method("getName")->willReturn("object-bridge-name");
+        $this->concrete = $this->createMock(Concrete::class);
+        $this->concrete->method("getId")->willReturn(5);
+    }
+
+    public function testDataArrayIdIsCorrectInConvertObjectBridgeConcretesToArray(): void
+    {
+        $array = $this->adapter->convertObjectBridgeConcretesToArray($this->objectBridge, $this->concrete);
+
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/DataResourceAdapterTest.php]: First check if $array[0] exists. If it doesn't PHP might raise an error that stops execution of the whole test suite (depending on how php and phpunit is configured).

File: tests/unit/ObjectBridgeService/DataResourceAdapterTest.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    public function testReturnEmptyArrayOnEmptyInputInConvertArraysToObjectBridgeConcretes(): void
+    {
+        $concretes = $this->adapter->convertArraysToObjectBridgeConcretes();
+
+        $this->assertEmpty($concretes);
+    }
+
+    public function testExceptionOnFailedToFindInstanceInConvertArraysToObjectBridgeConcretes(): void
+    {
```
Comment (JeroenVanDerLaan) [inline tests/unit/ObjectBridgeService/DataResourceAdapterTest.php]: Might want to add this fixture to the test setup, preventing some duplicate code.

File: src/ObjectBridgeService/DataResourceAdapter.php [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            $id = $concrete->getId();
+            $array = $this->createDataArray($id, $name, $index);
+            array_push($data, $array);
+            $index++;
+        }
+        return $data;
+    }
+
+    /**
+     * @param array ...$concreteDataArrays
```
Comment (JeroenVanDerLaan) [inline src/ObjectBridgeService/DataResourceAdapter.php]: Not a breaking issue, but PHPStorm unfortunately doesn't pick up array element types automatically. You could put `@return Concrete[]` here to ensure best IDEA integration. Up to you if you want to make this change.
